unattended-upgrade

```diff
@@ -1681,8 +1702,11 @@ def main(options, rootdir=""):
         # redo the selection about the packages to upgrade based on the new
         # blacklist
         logging.debug("blacklist: %s" % blacklisted_pkgs)
+        blacklist = [re.compile(pattern) for pattern in blacklisted_pkgs]
```

Comment:
This is nice! I wonder how much speed up this brings in itself.

Reply:
Well, actually nothing in practice, I just re-run the tests on AC power several times and the original measured speed-up was observed due to the low number of samples. Dropping this commit as it is not worth the added complexity.

Reply:
In the re-run the full speedup was 45s -> 4.4s with the commits up to the candidate adjustment changes.